build/build/src/httpbin.rs

```diff
@@ -49,23 +42,38 @@ impl Drop for Spawned {
     }
 }
 
-pub async fn get_and_spawn_httpbin_on_free_port() -> Result<Spawned> {
-    get_and_spawn_httpbin(ide_ci::get_free_port()?).await
+pub async fn get_and_spawn_httpbin_on_free_port(
+    sbt: &crate::engine::sbt::Context,
+) -> Result<Spawned> {
+    get_and_spawn_httpbin(sbt, ide_ci::get_free_port()?).await
```

Comment:
Wouldn't it be better, if the `sbt simple-httpbin/run localhost` command just randomly opened a port and printed it out? Checking a free port by `ide_ci::get_free_port()` and then using it is slightly fragile and subject to some other process occupying that port meanwhile... @hubertp

Reply:
The previous comment is based on an observation that it is now _our httpbin_ server and we can tailor it to our needs.

To amplify that, what about: let the JSON tests be aware of _our httpbin_ server and let the tests launch it and stop it themselves!?

Then we would need no magic in the Rust build system. Instead of using `ENSO_...HTTP...` property in the tests (which individuals usually don't provide) and skipping the tests when it is not set, the tests would always prepare their  _own httpbin_ server and run properly with it?

Reply:
I'm guessing our tests would need to be somehow aware of the httpbin project and I haven't seen that scenario yet in our test cases. So the packaging might be tricky to set up on that side. Other than that I don't see problems with introducing that suggestion.

The only advantage of using the env variable is that we can quickly swap a different httpbin, not necessarily the one written in Go but also in Python etc.

Reply:
> Wouldn't it be better, if the `sbt simple-httpbin/run localhost` command just randomly opened a port and printed it out? Checking a free port by `ide_ci::get_free_port()` and then using it is slightly fragile and subject to some other process occupying that port meanwhile... @hubertp

You are correct, current behavior is race-y. Should not cause trouble in CI, as each runner is a separate container, but still.

> The previous comment is based on an observation that it is now _our httpbin_ server and we can tailor it to our needs.
> 
> To amplify that, what about: let the JSON tests be aware of _our httpbin_ server and let the tests launch it and stop it themselves!?
> 
> Then we would need no magic in the Rust build system. Instead of using `ENSO_...HTTP...` property in the tests (which individuals usually don't provide) and skipping the tests when it is not set, the tests would always prepare their _own httpbin_ server and run properly with it?

Yes, having the tests take care of their own setup would be definitely much cleaner, nicer,

Reply:
> I'm guessing our tests would need to be somehow aware of the httpbin project and I haven't seen that scenario yet in our test cases. So the packaging might be tricky to set up on that side. 

CCing @radeusgd

Reply:
IMHO the easiest way would be to move the httpbin class completely into `test/Tests/polyglot` and then we can run the Java method thru Enso-Java Polyglot. We could then have a similar setup to `std-bits` and `std-table` compiling the polyglot extensions for the project.

Other approach we could do is have a similar setup to `std-bits` but instead put a compiled JAR into `test/Tests/data/`. Then we could run it as any other application through our `Process` API.

But IMHO running the test server directly through polyglot seems like a cleaner and more elegant solution.